src/context/PortfolioProvider/PortfolioContext.tsx

```diff
@@ -171,6 +182,79 @@ export const PortfolioProvider = ({ children }: { children: React.ReactNode }) =
     })()
   }, [assetsById, chainAdapter, dispatch, wallet])
 
+  const txIds = useSelector(selectTxIds)
+  const txsById = useSelector(selectTxs)
+  const txHistoryStatus = useSelector(selectTxHistoryStatus)
+
+  /**
+   * monitor for new pending txs, add them to a set, so we can monitor when they're confirmed
+   */
+  useEffect(() => {
+    // we only want to refetch portfolio if a new tx comes in after we're finished loading
+    if (txHistoryStatus !== 'loaded') return
+    // don't fire with nothing connected
+    if (isEmpty(accountSpecifiers)) return
+    // grab the most recent txId
+    const txId = head(txIds)!
+    // grab the actual tx
+    const tx = txsById[txId]
+    // always wear protection, or don't it's your choice really
+    if (!tx) return
+
+    if (tx.caip2 === cosmosChainId) {
```

Comment:
This is kinda hardcoding, I prefer using the tx status itself here, if tx is already confirmed, refetch the account and if it's pending, add it up to the `pendingTxIds`. this way there's no need to change this later if another chain is added to the app and has the same behavior as cosmos.